### ISSUE TITLE
valid-jsdoc @return check does not realize any async function returns a Promise

### ISSUE THREAD

**Issue by n-rook** [eslint/eslint#9881]
**Tell us about your environment**

* **ESLint Version:** v4.16.0
* **Node Version:** 8.9.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
'use strict';

module.exports = {
  extends: 'eslint:recommended',
  env: {
    es6: true,
    mocha: true,
    node: true
  },
  root: true,
  parserOptions: {
    ecmaVersion: 8,
  },
  rules: {
    'valid-jsdoc': ['error',
      {
        requireReturn: false,
        requireReturnDescription: false,
        requireParamDescription: false
      }
    ],
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/**
 * An example function which doesn't do anything.
 *
 * @return {!Promise}
 */
async function exampleFunction() {
  // Do nothing
}

console.log(exampleFunction() instanceof Promise);  // this prints "true"
```

```bash
$(npm bin)/eslint .
```

**What did you expect to happen?**
No valid-jsdoc error.

**What actually happened? Please include the actual, raw output from ESLint.**
This error:
`   4:4  error  Unexpected @return tag; function has no return statement  valid-jsdoc`

**Comment by not-an-aardvark** [comment]
Thanks for the report, I can reproduce this issue. It only seems to happen when the `requireReturn: false` option is used.

**Comment by rachx** [comment]
Working on the issue

**Comment by n-rook** [comment]
Nice! Thanks @rachx and code reviewers, I will not miss putting `// eslint-disable-next-line valid-jsdoc` everywhere :)
